packages/meditrak-app/app/assessment/selectors.js

```diff
@@ -109,3 +118,80 @@ export const getQuestion = (state, questionId) => state.assessment.questions[que
 export const getAnswers = state => state.assessment.answers;
 
 export const getAnswerForQuestion = (state, questionId) => getAnswers(state)[questionId];
+
+const getArithmeticResult = (state, config) => {
+  const { formula, valueTranslation, defaultValues = {}, text = '' } = config.calculated;
+  const values = {};
+  const variables = getVariables(formula);
+  let translatedText = text;
+
+  variables.forEach(questionIdVariable => {
+    let answer = getAnswerForQuestion(state, questionIdVariable) || 0;
+
+    if (
+      valueTranslation &&
+      valueTranslation[questionIdVariable] &&
+      valueTranslation[questionIdVariable][answer] !== undefined
+    ) {
+      answer = valueTranslation[questionIdVariable][answer];
+    }
+
+    if (answer === undefined) {
+      answer = defaultValues[questionIdVariable];
+    }
+
+    values[questionIdVariable] = answer || 0;
+
+    translatedText = translatedText.replace(questionIdVariable, answer);
```

Comment:
Can we do this in a separate loop please? Doing the 4 units of work 50 times is the same as doing 2 units of work 50 times plus another 2 units of work 50 times - i.e. splitting the contents of an iteration apart doesn't really slow things down (there's a tiny bit of overhead setting up the loop, but that's negligible). I think in this case the logic is distinct enough to justify a separate block. I'd also try to aim for `const` rather than `let` where possible (e.g. by making use of a function that contains the conditional logic with returns)